### ISSUE TITLE
It should be possible to clear cache even when the user is logged out

### ISSUE THREAD

**Issue by skel35** [toggl-open-source/toggldesktop#3897]
### 💻 Environment
Platform: macOS

### 📒 Description
It should be possible to clear the cache even when the user is logged out.
Currently, the "Clear cache" option in the Account menu is greyed out in this scenario.

**Comment by IndrekV** [comment]
The way clear cache works is it clears all the data of a certain user. It does not clear the whole database. That is why it is disabled while logging out. In that moment there is no info about the user data it should clear.

**Comment by IndrekV** [comment]
What is the scenario that demands this functionality?

**Comment by skel35** [comment]
The scenario was that there was a database error while the user is logged out which prevented the user from logging in. The only workaround for it was manually deleting the database file.
I didn't realize Clear Cache deletes only the current user data, now I'm not sure what's a good solution to this issue.

**Comment by IndrekV** [comment]
We could do it so that 
- if user is logged in it works the same as right now.
- if user is not logged in we ask "Do you really want the remove the local database" and then it will delete the database and restart the app.

What do you think?

**Comment by MartinBriza** [comment]
what's the rationale behind keeping the data for logged out users? I'd
assume a big portion of them expect the data to be gone when they log out

**Comment by NghiaTranUIT** [comment]
Totally agree with Martin, we should clear all cache when the user logout and it does in the background without any prompt.

**Comment by IndrekV** [comment]
The idea behind keeping data after logout was that the user could login while offline and then track time. I'm not 100% this works right now.

**Comment by IndrekV** [comment]
I think we should stick with the solution proposed in: 

> We could do it so that
> 
> * if user is logged in it works the same as right now.
> * if user is not logged in we ask "Do you really want the remove the local database" and then it will delete the database and restart the app.
> 
> What do you think?

**Comment by NghiaTranUIT** [comment]
I agree that Clear Cache in Logout would affect the offline mode, but I do not totally encourage to prompt and interrupt the user action the user's intention just simple logout.

If we prompt, it causes confusion because they don't know that there is a local database or not and why do they need to delete the local database under the normal users's point of view.

I still prefer TogglDesktop implicitly makes its decisions without prompting. 

Do you have any thought @skel35 ?

**Comment by IndrekV** [comment]
Prompting on logout should be avoided but I think we shouldn't clear user data at all on logout.
If we decide to go the route to delete user data on logout then I think there should still be a "Clear Cache" option in the menu that would then just delete the whole database. This is beneficial in cases of database errors so the users can solve the issue from inside the app and don't have to start looking for the correct directory to manually delete local database file.

**Comment by IndrekV** [comment]
The solution I proposed does not prompt on logout.

**Comment by skel35** [comment]
I like Indrek's solution, I think it nicely addresses the problem described in the issue.
This would basically allow the user to "Clear Cache" twice - first time would log out and clear this user's data and a second time would remove the entire database file.
The only thing I'm a bit concerned about is that the same button doing different things is a bit confusing for the user, we should find another name for one of them.

**Comment by IndrekV** [comment]
We could have different text in the menu item

If user is logged in - `Clear Cache`
If user is logged out - `Clear local database`

**Comment by skel35** [comment]
Sounds good.
I've updated the issue description with the results of this discussion.
